### DISCUSSION TITLE
PCAN adapter problem

### DISCUSSION THREAD

**Discussion by rcd09** [canopen-python/canopen#336] [Q&A]
Hi all,

Great package.  I would like to write a program that would listen for activity on a CANopen network.  If none or the network is not up at the time, it would wait several seconds and try until there is and then do something.  Basically the code:

try:
1. Instantiates a can.Network
2. Calls network.connect()
3. Calls network.scanner.search()
4. Catch CanInitializationError and CanInterfaceNotImplementedError if raised.  All these do is print a message.
5. If nothing found, wait five seconds and go to 1.

This is fine with my Vector box - handles the non-existant network just fine and when it is there connects and searches without issue.

But my pcan adapter ends up with "Can Error: adapter 'pcan' and/or channel 'PCAN_USBBUS1' could not be initialized." and I can't get rid of that until I reboot!  I guess the pcan driver is stuck in some thread or exception somewhere.

Anybody know what I am doing wrong or what I can do to recover a pcan adapter after attempting to connect and search when there is no CAN network there?

**Comment by acolomb** [comment]
Which operating system are you on? How did you install the PCAN driver?

**Comment by rcd09** [comment]
D'oh!  Of course.  Windows 10 and PCAN-USB driver 4.1.4.16279 from 2018.
Let me update that and get back to you if its still a problem.
